pkg/beeclient/api/bytes.go

```diff
@@ -22,7 +22,11 @@ type BytesUploadResponse struct {
 }
 
 // Upload uploads bytes to the node
-func (b *BytesService) Upload(ctx context.Context, data io.Reader) (resp BytesUploadResponse, err error) {
-	err = b.client.request(ctx, http.MethodPost, "/"+apiVersion+"/bytes", data, &resp)
+func (b *BytesService) Upload(ctx context.Context, data io.Reader, pin bool) (resp BytesUploadResponse, err error) {
```

Comment:
Pin argument is optional and it would be better to have it as the part of the options struct passed to this method as the last argument. This allows other options (like tag and encrypt) to be easily added instead to change the function signature.